### ISSUE TITLE
Execution hangs after particular step in CUDA 10.1 TF 1.14.0

### ISSUE THREAD

**Issue by sakshambanga** [tensorflow/tensorflow#35310]
### System information
- Have I written custom code: Yes
- OS Platform and Distribution: 18.04.3 LTS Ubuntu
- TensorFlow installed : pre installed container
- TensorFlow version : 1.14.0
- Python version: 3.5.2
- Bazel version (if compiling from source): - 
- GCC/Compiler version (if compiling from source): 7.4.0
- CUDA/cuDNN version: 10.1
- GPU model and memory: Tesla V100-SXM3-32GB
- Exact command to reproduce: Running the command python train_blstm.py

### Describe the problem
I am working on an Bi-LSTM + CTC Loss + WordBeamSearch Architecture for online handwriting recognition. The problem is that the code hangs/stalls after a particular point everytime. I have also let it run for 2 days now but it's still the same. Any help in this regard would be appreciated.

### Source code / logs
Here is the traceback of the issue attached below:
```

2019-12-20 14:18:51.291556: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.1
W1220 14:18:54.480478 139982147983168 deprecation_wrapper.py:119] From train_blstm_dec_19.py:295: The name tf.app.run is deprecated. Please use tf.compat.v1.app.run instead.

W1220 14:18:54.481328 139982147983168 deprecation_wrapper.py:119] From train_blstm_dec_19.py:155: The name tf.get_default_graph is deprecated. Please use tf.compat.v1.get_default_graph instead.

data_dir: ./
checkpoints_dir: ../checkpoints/
log_dir1: ../train_log/
restore_path: None
batch_size: 128
total_epoches: 300
hidden_size: 128
num_layers: 2
input_dims: 10
num_classes: 80
save_freq: 5
learning_rate: 0.001
decay_rate: 0.99
momentum: 0.9
max_length: 1940.0
label_pad: 63
if_valid_vr: False
W1220 14:18:59.086257 139982147983168 deprecation_wrapper.py:119] From /workspace/data/src_hit_word_beam_edition/model_blstm_dec_19.py:30: The name tf.train.get_or_create_global_step is deprecated. Please use tf.compat.v1.train.get_or_create_global_step instead.

W1220 14:18:59.098966 139982147983168 deprecation_wrapper.py:119] From /workspace/data/src_hit_word_beam_edition/model_blstm_dec_19.py:31: The name tf.placeholder is deprecated. Please use tf.compat.v1.placeholder instead.

W1220 14:18:59.106423 139982147983168 deprecation.py:323] From /workspace/data/src_hit_word_beam_edition/model_blstm_dec_19.py:36: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where
W1220 14:18:59.112254 139982147983168 deprecation_wrapper.py:119] From /workspace/data/src_hit_word_beam_edition/model_blstm_dec_19.py:113: The name tf.variable_scope is deprecated. Please use tf.compat.v1.variable_scope instead.

W1220 14:18:59.112922 139982147983168 deprecation_wrapper.py:119] From /workspace/data/src_hit_word_beam_edition/model_blstm_dec_19.py:116: The name tf.get_variable_scope is deprecated. Please use tf.compat.v1.get_variable_scope instead.

W1220 14:18:59.113215 139982147983168 deprecation.py:323] From /workspace/data/src_hit_word_beam_edition/model_blstm_dec_19.py:116: LSTMCell.__init__ (from tensorflow.python.ops.rnn_cell_impl) is deprecated and will be removed in a future version.
Instructions for updating:
This class is equivalent as tf.keras.layers.LSTMCell, and will be replaced by that in Tensorflow 2.0.
W1220 14:18:59.115090 139982147983168 deprecation.py:323] From /usr/local/lib/python3.6/dist-packages/tensorflow/contrib/rnn/python/ops/rnn.py:239: bidirectional_dynamic_rnn (from tensorflow.python.ops.rnn) is deprecated and will be removed in a future version.
Instructions for updating:
Please use `keras.layers.Bidirectional(keras.layers.RNN(cell))`, which is equivalent to this API
W1220 14:18:59.115478 139982147983168 deprecation.py:323] From /usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/rnn.py:464: dynamic_rnn (from tensorflow.python.ops.rnn) is deprecated and will be removed in a future version.
Instructions for updating:
Please use `keras.layers.RNN(cell)`, which is equivalent to this API
W1220 14:18:59.192415 139982147983168 deprecation.py:506] From /usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/init_ops.py:1251: calling VarianceScaling.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.
Instructions for updating:
Call initializer instance with the dtype argument instead of passing it to the constructor
W1220 14:18:59.200585 139982147983168 deprecation.py:506] From /usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/rnn_cell_impl.py:961: calling Zeros.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.
Instructions for updating:
Call initializer instance with the dtype argument instead of passing it to the constructor
stack_bidirectional_dynamic_rnn: Tensor("blstm/Reshape:0", shape=(?, 1940, 2, 128), dtype=float32)
W1220 14:19:00.570831 139982147983168 deprecation_wrapper.py:119] From /workspace/data/src_hit_word_beam_edition/model_blstm_dec_19.py:122: The name tf.truncated_normal is deprecated. Please use tf.random.truncated_normal instead.

Tensor("blstm/unstack:0", shape=(?, 2, 128), dtype=float32)
Tensor("blstm/add:0", shape=(?, 128), dtype=float32)
Tensor("blstm/stack:0", shape=(1940, ?, 80), dtype=float32)
Tensor("blstm/stack:0", shape=(1940, ?, 80), dtype=float32)
W1220 14:19:08.754076 139982147983168 deprecation_wrapper.py:119] From /workspace/data/src_hit_word_beam_edition/model_blstm_dec_19.py:87: The name tf.nn.ctc_loss is deprecated. Please use tf.compat.v1.nn.ctc_loss instead.

W1220 14:19:08.757078 139982147983168 deprecation_wrapper.py:119] From /workspace/data/src_hit_word_beam_edition/model_blstm_dec_19.py:89: The name tf.summary.scalar is deprecated. Please use tf.compat.v1.summary.scalar instead.

Tensor("ctc_loss/Mean:0", shape=(), dtype=float32)
W1220 14:19:08.758585 139982147983168 deprecation_wrapper.py:119] From /workspace/data/src_hit_word_beam_edition/model_blstm_dec_19.py:93: The name tf.train.AdamOptimizer is deprecated. Please use tf.compat.v1.train.AdamOptimizer instead.

W1220 14:20:39.822741 139982147983168 deprecation.py:506] From /workspace/data/src_hit_word_beam_edition/model_blstm_dec_19.py:101: calling softmax (from tensorflow.python.ops.nn_ops) with dim is deprecated and will be removed in a future version.
Instructions for updating:
dim is deprecated, use axis instead
W1220 14:20:39.824999 139982147983168 deprecation_wrapper.py:119] From /workspace/data/src_hit_word_beam_edition/model_blstm_dec_19.py:41: The name tf.summary.merge_all is deprecated. Please use tf.compat.v1.summary.merge_all instead.

W1220 14:20:39.827233 139982147983168 deprecation_wrapper.py:119] From train_blstm_dec_19.py:203: The name tf.train.Saver is deprecated. Please use tf.compat.v1.train.Saver instead.

2019-12-20 14:20:39.892933: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcuda.so.1
2019-12-20 14:20:40.039163: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: 
name: Tesla V100-SXM3-32GB major: 7 minor: 0 memoryClockRate(GHz): 1.597
pciBusID: 0000:be:00.0
2019-12-20 14:20:40.039210: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.1
2019-12-20 14:20:40.041126: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10
2019-12-20 14:20:40.042838: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10
2019-12-20 14:20:40.043132: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10
2019-12-20 14:20:40.044996: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10
2019-12-20 14:20:40.046033: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10
2019-12-20 14:20:40.049918: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7
2019-12-20 14:20:40.056223: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0
2019-12-20 14:20:40.096725: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2700000000 Hz
2019-12-20 14:20:40.146344: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0xf2c8a20 executing computations on platform Host. Devices:
2019-12-20 14:20:40.146400: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>
2019-12-20 14:20:40.580239: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0xcd331c0 executing computations on platform CUDA. Devices:
2019-12-20 14:20:40.580287: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): Tesla V100-SXM3-32GB, Compute Capability 7.0
2019-12-20 14:20:40.583875: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: 
name: Tesla V100-SXM3-32GB major: 7 minor: 0 memoryClockRate(GHz): 1.597
pciBusID: 0000:be:00.0
2019-12-20 14:20:40.583911: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.1
2019-12-20 14:20:40.583946: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10
2019-12-20 14:20:40.583968: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10
2019-12-20 14:20:40.583978: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10
2019-12-20 14:20:40.583987: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10
2019-12-20 14:20:40.583997: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10
2019-12-20 14:20:40.584007: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7
2019-12-20 14:20:40.591236: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0
2019-12-20 14:20:40.591303: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.1
2019-12-20 14:20:41.297763: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-12-20 14:20:41.297816: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 
2019-12-20 14:20:41.297824: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N 
2019-12-20 14:20:41.305712: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 30466 MB memory) -> physical GPU (device: 0, name: Tesla V100-SXM3-32GB, pci bus id: 0000:be:00.0, compute capability: 7.0)
2019-12-20 14:27:51.088076: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10

```

**Comment by gadagashwini** [comment]
@sakshambanga, Could you provide the standalone code to reproduce the reported issue. Thanks!

**Comment by gadagashwini** [comment]
@sakshambanga, Any update on code snippet!

**Comment by gadagashwini** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
